src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -253,16 +264,50 @@ public DetailAST getUnderlyingNode() {
     }
 
     /**
-     * Finds child element with {@link TokenTypes#IDENT}, extracts its value and stores it.
-     * Value can be accessed using {@code @text} attribute. Now {@code @text} attribute is only
-     * supported attribute.
+     * Finds child element with {@link TokenTypes#IDENT},
+     * extracts its value, creates {@code AttributeNode} object and returns it.
+     * Value can be accessed using {@code @text} attribute.
+     *
+     * @return attribute object
      */
-    private void setIdent() {
-        final DetailAST identAst = detailAst.findFirstToken(TokenTypes.IDENT);
-        if (identAst != null) {
-            ident = identAst.getText();
+    private AbstractNode getIdent() {
+        AttributeNode attribute = null;
+        if (detailAst.getChildCount(TokenTypes.IDENT) == 1) {
+            ident = detailAst.findFirstToken(TokenTypes.IDENT).getText();
+            attribute = new AttributeNode(TEXT_ATTRIBUTE_NAME, ident);
+        }
+        return attribute;
+    }
+
+    /**
+     * Checks if current node is {@link TokenTypes#IDENT},
+     * extracts its value, creates {@code AttributeNode} object and returns it.
+     * Value can be accessed using {@code @value} attribute.
+     *
+     * @return attribute object
+     */
+    private AbstractNode getValue() {
+        AttributeNode attribute = null;
+        if (detailAst.getType() == TokenTypes.IDENT) {
+            value = detailAst.getText();
+            attribute = new AttributeNode(VALUE_ATTRIBUTE_NAME, detailAst.getText());
+        }
+        return attribute;
+    }
+
+    /**
+     * Initializes array of attributes.
+     */
+    private void setAttributes() {
+        final AbstractNode identAttribute = getIdent();
+        final AbstractNode valueAttribute = getValue();
+        if (identAttribute != null) {
+            attributes = new AbstractNode[1];
```

Comment:
please move `attributes = new AbstractNode[1];` above IFs.

Reply:
I'm not sure about the checkstyle's usage of arrays vs. collections. Instead of using `attributes`, `value` and `ident` using `Map<String, AbstractNode> attributes` should help to simplify (a) access to available attributes at `getAttributeValue(...)`, (b) values can be returned as array inside `iterateAxis(...)` and (c) allow adding multiple attributes just as needed in this place.

Reply:
@lars-sh thanks, edited code using your PR